sdk/core/Azure.Core/samples/DynamicContent.md

```diff
@@ -0,0 +1,166 @@
+
+```C# Snippet:AzureCoreGetDynamicJsonArrayValue
+Response response = client.GetWidget();
+dynamic widget = response.Content.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+// JSON is `{ "values" : [1, 2, 3] }`
+if (widget.Values.Length > 0)
+{
+    int value = widget.Values[0];
+}
+```
+
+### Check whether an optional property is present
+
+Optional properties will return null if not present in the JSON content.
+
+```C# Snippet:AzureCoreGetDynamicJsonOptionalProperty
+Response response = client.GetWidget();
+dynamic widget = response.Content.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+// JSON is `{ "details" : { "color" : "blue", "size" : "small" } }`
+
+// Check whether optional property is present
+if (widget.Details != null)
+{
+    string color = widget.Details.Color;
+}
+```
+
+### Enumerate a collection
+
+Dynamic JSON objects and arrays are `IEnumerable` and can be iterated over with the `foreach` keyword.
+
+```C# Snippet:AzureCoreEnumerateDynamicJsonObject
+Response response = client.GetWidget();
+dynamic widget = response.Content.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+// JSON is `{ "details" : { "color" : "blue", "size" : "small" } }`
+foreach (dynamic property in widget.Details)
+{
+    Console.WriteLine($"Widget has property {property.Name}='{property.Value}'.");
+}
+```
+
+### Get a property with invalid C# characters in the name
+
+JSON members whose names have characters that are not valid for property names in C# can be accessed using property indexers.
+
+```C# Snippet:AzureCoreGetDynamicPropertyInvalidCharacters
+Response response = client.GetWidget();
+dynamic widget = response.Content.ToDynamicFromJson();
+
+/// JSON is `{ "$id" = "123" }`
+string id = widget["$id"];
+```
+
+### Cast to a POCO type
+
+Dynamic JSON objects can be cast to CLR types using the cast operator.
+
+```C# Snippet:AzureCoreCastDynamicJsonToPOCO
+Response response = client.GetWidget();
+dynamic content = response.Content.ToDynamicFromJson(DynamicCaseMapping.PascalToCamel);
+
+// JSON is `{ "id" : "123", "name" : "Widget" }`
+Widget widget = (Widget)content;
+```
+
+```C# Snippet:AzureCoreDynamicJsonPOCO
+public class Widget
+{
+    public string Id { get; set; }
+    public string Name { get; set; }
+}
+```
+
+### Working with Azure values
+
+When working with JSON from Azure services, you can learn what properties are available in the JSON response content from the REST API documentation for the service, examples in the protocol method documentation, or by expanding the [Dynamic View](https://learn.microsoft.com/visualstudio/debugger/watch-and-quickwatch-windows) in Visual Studio.
+
+To behave as much as possible like Azure SDK model types, `DynamicData` allows JSON members to be accessed using PascalCase property names, and will write any added properties with camelCase names.  If there is a need to bypass these name mappings, JSON members can be accessed with exact strings using property indexers.
```

Comment:
This paragraph no longer seems correct since we default to unchanged case. Your snippet shows this, but doesn't seem to align with the text - notably the first sentence.

Reply:
Cool, yeah, I was just rehydrating the old thing to update later, but I can fix this as part of this PR.